src/features/user/Account/History.tsx

```diff
@@ -60,6 +78,46 @@ export default function History({
     }
   }, [paymentHistory]);
 
+  const handleIssueReceipts = async () => {
+    setLoading(true);
+    try {
+      const res = await postAuthenticatedRequest(
+        '/app/taxReceipts',
+        {
+          year: new Date().getFullYear(),
+        },
+        token,
+        handleError
+      );
+      setLoading(false);
+      if (res && res.length === 0) {
+        setError({
+          type: 'error',
+          message: t('me:taxReceiptsAlreadyGenerated'),
+        });
+        setSelectedRecord(0);
+      } else {
+        await fetchPaymentHistory();
+        setSelectedRecord(0);
+        setOpen(true)
+      }
+    } catch (err) {
+      setLoading(false);
+      console.error(err);
+    }
+  };
```

Comment:
I simulated an error by changing the API url to an incorrect URL, resulting in a 404 not found response from the API. 

I see an error "Requested page was not found" as well as a success snackbar in this case, followed by the page reloading.

How would unexpected API responses be handled here.

Reply:
In ErrorHandling PR. there is a check for non 200 response i.e
https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/2d3569dc5fa3d2ea1ec07ad89314e120348313ca/src/utils/apiRequests/api.ts#L208

It will check for we get success(i.e 200 response) or error (i.e 400 and  500 series response). This will take care of the above scenario. 

Since error handling wasn't a part of develop yet I have kept it as it is.